src/app/Shared/Services/Report.service.tsx

```diff
@@ -101,13 +101,102 @@ export class ReportService {
     );
   }
 
+  reportJson(recording: Recording, connectUrl: string): Observable<[string, RuleEvaluation][]> {
+    if (!recording?.reportUrl) {
+      return throwError(() => new Error('No recording report URL'));
+    }
+    return this.login.getHeaders().pipe(
+      concatMap((headers) => {
+        headers.append('Accept', 'application/json');
+        return fromFetch(recording.reportUrl, {
+          method: 'GET',
+          mode: 'cors',
+          credentials: 'include',
+          headers,
+        });
+      }),
+      concatMap(async (resp) => {
+        if (resp.ok) {
+          const obj = JSON.parse(await resp.text());
+          return Object.entries(obj) as [string, RuleEvaluation][];
+        } else {
+          const ge: GenerationError = {
+            name: `Report Failure (${recording.name})`,
+            message: resp.statusText,
+            messageDetail: from(resp.text()),
+            status: resp.status,
+          };
+          throw ge;
+        }
+      }),
+      tap({
+        next: (report) => {
+          const isArchived = !isActiveRecording(recording);
+          const isActiveStopped = isActiveRecording(recording) && recording.state === RecordingState.STOPPED;
+          if (isArchived || isActiveStopped) {
+            try {
+              sessionStorage.setItem(this.analysisKeyTimestamp(connectUrl), Date.now().toString());
+              sessionStorage.setItem(this.analysisKey(connectUrl), JSON.stringify(report));
+            } catch (error) {
+              this.notifications.warning('Report Caching Failed', (error as any).message);
+              sessionStorage.clear();
```

Comment:
Should this just remove the entries, applicable for report cache? I think web console credentials are also saved in sessionStorage so they will cleared as well and requires user to log in again.

Reply:
Should call `this.deleteCachedAnalysisReport(connectUrl);`?

Reply:
Ah, right. Forgot about this, thanks.